cmake/FindSCIPY_OPENBLAS32.cmake

```diff
@@ -0,0 +1,40 @@
+###############################################################################################
+# This file provides macros to process CUDA, CUDAToolkit, cuQuantum and MPI external libraries.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+macro(get_scipy_openblas_libname SCIPY_OPENBLAS_LIB_NAME)
+    set(LIB_SUFFIX ".so")
+    if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+        set(LIB_SUFFIX ".dylib")
+    elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
+        set(LIB_SUFFIX ".dll")
+    endif()
+
+    set(${SCIPY_OPENBLAS_LIB_NAME} "libscipy_openblas${LIB_SUFFIX}")
+endmacro()
+
+macro(find_path_to_openblas SCIPY_OPENBLASE_LIB_PATH)
+    set(SCIPY_OPENBLAS_LIB_NAME "")
+    get_scipy_openblas_libname(SCIPY_OPENBLAS_LIB_NAME)
+
+    find_file(SCIPY_OPENBLAS_LIB_FILE
+        NAMES   ${SCIPY_OPENBLAS_LIB_NAME}
+        HINTS   ${SCIPY_OPENBLAS}/lib
+                $ENV{SCIPY_OPENBLAS32}/lib
+                ${Python_SITELIB}/scipy_openblas32/lib
+                ${SCIPY_OPENBLAS}
+                $ENV{SCIPY_OPENBLAS32}
```

Comment:
Too many hints! :smile:  I know you've spent some time on gathering them all :rocket: